### PR TITLE
feat: hint support render react node

### PR DESCRIPTION
**Pull request by superwf**
**Describe the pull request**

A clear and concise description of what is being fixed, added or modified.

**Why**

Why are these changes necessary?

**How**

How were these changes implemented?

**Screenshots**

If applicable, add screenshots to help explain what is being modified.

### CODE REVIEW

File: src/components/Hints/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1,3 +1,5 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`Hints should render jsx 1`] = `null`;
+
```
Comment (HiDeoo) [inline src/components/Hints/__snapshots__/index.test.js.snap:4]: I don't think this snapshot is very useful for this specific case.

File: src/components/Hints/index.test.js [hunk 2]
```diff
@@ -23,6 +23,24 @@ describe('Hints', () => {
...
+            {
+              element: '.test',
+              hint: <b>test</b>,
+            },
+          ]}
+        />
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
```
Comment (HiDeoo) [inline src/components/Hints/index.test.js:41]: Looks like the snapshot is `null` so it's not really useful in this case, maybe we should do something like [the equivalent test](https://github.com/HiDeoo/intro.js-react/blob/24d924f1fd69fa3cec984f4903725d0b536b3f2e/src/components/Steps/index.test.js#L29-L43) for the steps and check that at least `renderToStaticMarkup()` is even called.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Additionally, the TypeScript definition for the `hint` property of the `Hint` interface should be updated https://github.com/HiDeoo/intro.js-react/blob/24d924f1fd69fa3cec984f4903725d0b536b3f2e/index.d.ts#L40 

to something like:

```ts
hint: string | React.ReactNode;
```

File: src/components/Hints/__snapshots__/index.test.js.snap (hunk 1 above)
Comment (mannyad) [reply src/components/Hints/__snapshots__/index.test.js.snap:4]: Okay

**Comment by HiDeoo** [comment]
I'm going to close this pull request due to inactivity. Thanks for your contribution nonetheless.

**Pull request closed without merging**